File: solutions/python3/0056.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        ret = []
        intervals.sort()

        for interval in intervals:
            if not ret or ret[-1][1] < interval[0]:
                ret.append(interval)
            else:
                ret[-1][1] = max(ret[-1][1], interval[1])

        return ret

File: solutions/cpp/0056.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> ret;
        sort(intervals.begin(), intervals.end(), compare);

        for (const auto& interval : intervals) {
            if (ret.empty() || ret.back()[1] < interval[0])
                ret.push_back(interval);
            else
                ret.back()[1] = max(ret.back()[1], interval[1]);
        }

        return ret;
    }

private:
    bool static compare(const vector<int>& a, const vector<int>& b) {
        return a[0] < b[0];
    }
};